ingenialink/servo.py

```diff
@@ -273,6 +275,64 @@ def is_listener_started(self) -> bool:
         """
         return self.__listener_servo_status is not None
 
+    def check_configuration(self, config_file, subnode: Optional[int] = None) -> None:
+        """Check if the drive is configured in the same way as the given configuration file.
+        Compares the value of each register in the given file with the corresponding value in the
+        drive.
+
+        Args:
+            config_file: the configuration to check
+            subnode: Subnode of the axis. Defaults to None.
+
+        Raises:
+            FileNotFoundError: If the configuration file cannot be found.
+            ValueError: If a configuration file from a subnode different from 0
+                is attempted to be loaded to subnode 0.
+            ValueError: If an invalid subnode is provided.
+            ILConfigurationError: If the configuration file differs from the drive state.
+
+        """
+        if subnode is not None and (not isinstance(subnode, int) or subnode < 0):
+            raise ValueError("Invalid subnode")
+        _, registers = self._read_configuration_file(config_file)
+
+        dest_subnodes = set(int(element.attrib["subnode"]) for element in registers)
+        if subnode == 0 and subnode not in dest_subnodes:
+            raise ValueError(f"Cannot load {config_file} to subnode {subnode}")
+        cast_data = {"float": float, "str": str}
+        registers_errored = []
```

Comment:
I would say mismatch.

Reply:
I'm using the same array to collect error messages when a ILError occurs while trying to read the register, so renaming it to "registers_mismatched" would not be quite correct for some cases.